assignments/pull_requests_test.go

```diff
@@ -0,0 +1,163 @@
+			if diff := cmp.Diff(wantTeacher, gotTeacher, protocmp.Transform()); diff != "" {
+				t.Errorf("getNextReviewer() mismatch (-wantTeacher, +gotTeacher):\n%s", diff)
+			}
+		}
+
+		// Adding a new teacher.
+		// Teacher is expected to be picked as reviewer len(teachers)-1 times.
+		wantTeacher := &pb.User{ID: 6}
+		teachers = append(teachers, wantTeacher)
+		for i := 0; i < len(teachers)-1; i++ {
+			gotTeacher, err := getNextReviewer(ID, teachers, teacherReviewCounter)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(wantTeacher, gotTeacher, protocmp.Transform()); diff != "" {
+				t.Errorf("getNextReviewer() mismatch (-wantTeacher, +gotTeacher):\n%s", diff)
+			}
+		}
+		teachers = teachers[:len(teachers)-1]
+
+		for i := 0; i < len(students)*3; i++ {
+			gotStudent, err := getNextReviewer(ID, students, groupReviewCounter)
+			if err != nil {
+				t.Fatal(err)
+			}
+			wantStudent := students[i%len(students)]
+			if diff := cmp.Diff(wantStudent, gotStudent, protocmp.Transform()); diff != "" {
+				t.Errorf("getNextReviewer() mismatch (-wantStudent, +gotStudent):\n%s", diff)
+			}
+		}
+
+		// Adding a new student
+		// Student is expected to be picked as reviewer len(student)-1 times.
+		wantStudent := &pb.User{ID: 4}
+		students = append(students, wantStudent)
+		for i := 0; i < len(students)-1; i++ {
+			gotStudent, err := getNextReviewer(ID, students, groupReviewCounter)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(wantStudent, gotStudent, protocmp.Transform()); diff != "" {
+				t.Errorf("getNextReviewer() mismatch (-wantStudent, +gotStudent):\n%s", diff)
+			}
+		}
+		students = students[:len(students)-1]
+	}
+}
+
+// TestPublishFeedbackComment tests creating a feedback comment on a pull request, with the given result.
+func TestPublishFeedbackComment(t *testing.T) {
+	qfTestOrg := scm.GetTestOrganization(t)
+	accessToken := scm.GetAccessToken(t)
+	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+	results := &score.Results{
+		Scores: []*score.Score{
+			{TestName: "Test1", TaskName: "1", Score: 5, MaxScore: 7, Weight: 2},
+			{TestName: "Test2", TaskName: "1", Score: 3, MaxScore: 9, Weight: 3},
+			{TestName: "Test3", TaskName: "1", Score: 8, MaxScore: 8, Weight: 5},
+			{TestName: "Test4", TaskName: "1", Score: 2, MaxScore: 5, Weight: 1},
+			{TestName: "Test5", TaskName: "1", Score: 5, MaxScore: 7, Weight: 1},
+			{TestName: "Test6", TaskName: "2", Score: 5, MaxScore: 7, Weight: 1},
+			{TestName: "Test7", TaskName: "3", Score: 5, MaxScore: 7, Weight: 1},
+		},
+	}
+	body := CreateFeedbackComment(results, &pb.Task{Name: "lab1/1"}, &pb.Assignment{ScoreLimit: 80})
+	// TODO(espeland): Remember to reset these when done testing
+	opt := &scm.IssueCommentOptions{
+		Organization: qfTestOrg,
+		Repository:   "oleespe-labs",
+		Body:         body,
+	}
+	if err := s.EditIssueComment(context.Background(), 1117670404, opt); err != nil {
```

Comment:
Couldn't we just use `scm.GetIssues()` command to get the issue number?

Reply:
That is the commentID, not issue number.
I will think about a way to make this test more approachable.

Reply:
You created a todo assigned to yourself for this, so closing this comment.